sql/parser/builtins.go

```diff
@@ -347,6 +348,80 @@ var builtins = map[string][]builtin{
 	"max": aggregateImpls(boolType, intType, floatType, stringType),
 	"min": aggregateImpls(boolType, intType, floatType, stringType),
 	"sum": aggregateImpls(intType, floatType),
+
+	// Timestamp/Date functions.
+
+	"age": {
+		builtin{
+			types:      typeList{timestampType},
+			returnType: DummyInterval,
+			fn: func(args DTuple) (Datum, error) {
+				return DInterval{Duration: time.Now().Sub(args[0].(DTimestamp).Time)}, nil
+			},
+		},
+		builtin{
+			types:      typeList{timestampType, timestampType},
+			returnType: DummyInterval,
+			fn: func(args DTuple) (Datum, error) {
+				return DInterval{Duration: args[0].(DTimestamp).Sub(args[1].(DTimestamp).Time)}, nil
+			},
+		},
+	},
+
+	"current_date": {
+		builtin{
+			types:      typeList{},
+			returnType: DummyDate,
+			fn: func(args DTuple) (Datum, error) {
+				return DDate{Time: time.Now().Truncate(24 * time.Hour)}, nil
+			},
+		},
+	},
+
+	"clock_timestamp":   {nowBuiltin()},
+	"current_timestamp": {nowBuiltin()},
+	"localtimestamp":    {nowBuiltin()},
```

Comment:
`clock_timestamp` and `current_timestamp` are not synonymous (in postgresql). `current_timestamp` (and `now`) is a value chosen once per transaction (not per statement!); `clock_timestamp` is the instantaneous value which may change during execution (like mysql's `sysdate()`). Assuming we change `now()` to be a fixed time we may want to have `clock_timestamp`/`sysdate()` with a separate implementation (although I've never seen an actual use of either of these functions in the wild)